File: frontend/control-pagos-web/src/types/deuda.ts
export type TipoDeuda = 'Recibido' | 'Realizado';

export type EstadoDeuda = 'AlDia' | 'Proximo' | 'Vencido' | 'Pagada' | 'Lejana';

export interface Pago {
  id: number;
  fecha: string;
  monto: number;
  nota: string | null;
}

export interface PagoCreateInput {
  fecha: string;
  monto: number;
  nota?: string | null;
}

export interface DeudaListItem {
  id: number;
  nombre: string;
  tipo: TipoDeuda;
  fechaFinPropuesta: string;
  saldoInicial: number;
  saldoPagado: number;
  saldoRemanente: number;
  cuotaMensual: number;
  porcentajeAvance: number;
  estado: EstadoDeuda;
}

export interface DeudaDetail extends DeudaListItem {
  fechaInicio: string;
  fechaFinProyectada: string | null;
  pagos: Pago[];
}

export interface DeudaFormInput {
  nombre: string;
  tipo: TipoDeuda;
  fechaInicio: string;
  fechaFinPropuesta: string;
  saldoInicial: number;
  cuotaMensual: number;
}

export interface DashboardResumen {
  totalPorCobrar: number;
  totalPorPagar: number;
  cantidadVencidas: number;
  cantidadProximas: number;
}

File: backend/src/ControlPagos.Data/Entities/Deuda.cs
namespace ControlPagos.Data.Entities;

public class Deuda
{
    public int Id { get; set; }
    public string Nombre { get; set; } = default!;
    public TipoDeuda Tipo { get; set; }
    public DateOnly FechaInicio { get; set; }
    public DateOnly FechaFinPropuesta { get; set; }
    public decimal SaldoInicial { get; set; }
    public decimal CuotaMensual { get; set; }
    public decimal SaldoPagado { get; set; }
    public decimal SaldoRemanente { get; private set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<Pago> Pagos { get; set; } = new List<Pago>();
}
